src/libs/setShouldShowComposeInputKeyboardAware/index.native.ts

```diff
@@ -1,12 +1,18 @@
 import {EmitterSubscription, Keyboard} from 'react-native';
+import getOperatingSystem from '@libs/getOperatingSystem';
 import * as Composer from '@userActions/Composer';
+import CONST from '@src/CONST';
 import SetShouldShowComposeInputKeyboardAware from './types';
 
-let keyboardDidHideListener: EmitterSubscription | null = null;
+let keyboardEventListener: EmitterSubscription | null = null;
+// On iOS, there is a visible delay in displaying input after the keyboard has been closed with the `keyboardDidHide` event
+// Because of that - on iOS we can use `keyboardWillHide` that is not available on android
+const keyboardEvent = getOperatingSystem() === CONST.OS.IOS ? 'keyboardWillHide' : 'keyboardDidHide';
```

Comment:
The proper way of writing cross-platform code is to use specific file extensions. I would suggest creating a module that exports the keyboard event name to use on each platform. For example:

1. `keyboardEventName/index.ios.js` exports 'keyboardWillHide'
1. `keyboardEventName/index.android.js` exports 'keyboardDidHide'

Reply:
Here we have different implementations for web and native. The only difference in the native platforms is this event. 
So I think that module should return a function, not an event name. To follow Your solution I need to create an abstraction: a function that returns the current function with a proper event, something like
```
const setShouldShowComposeInputKeyboardAwareBuilder = eventName => (shouldShow) => {
....
}
```
and then 
```
//index.ios.ts
export default setShouldShowComposeInputKeyboardAwareBuilder('keyboardWillHide);

//index.native.ts
export default setShouldShowComposeInputKeyboardAwareBuilder('keyboardDidHide);
```

Do You think is worth to complex code when we can keep this difference between platforms in one line like now?

Reply:
Exporting a function is fine as well, good suggestion. This is definitely a non-negotiable. You can read more about this in the readme: https://github.com/Expensify/App#philosophy (point 5 of that section).

Reply:
Done